### ISSUE TITLE
Hidden directories (starting with `.`) are silently skipped during scanning

### ISSUE THREAD

**Issue by davidpoblador** [rvben/rumdl#102]
## Description

When running `rumdl check` without explicit paths, the tool silently skips hidden directories (those starting with `.`), even when they contain markdown files that should be linted.

## Reproduction Steps

1. Create a hidden directory with markdown files:
```bash
mkdir .documentation
echo "# Test" > .documentation/test.md
```

2. Run rumdl check:
```bash
rumdl check
# or
rumdl check .
```

**Result:** The `.documentation/test.md` file is not processed.

3. Run with explicit path:
```bash
rumdl check .documentation
```

**Result:** The file is now processed correctly.

## Expected Behavior

Hidden directories should be included in the scan by default, or there should be a flag (e.g., `--all`, `--include-hidden`) to explicitly include them.

## Actual Behavior

Hidden directories are silently skipped with no indication to the user that files are being excluded.

## Configuration Attempts

I've tried the following without success:
- Setting `respect_gitignore = false` in `.rumdl.toml`
- Adding include patterns like `include = [".documentation/**/*.md", "**/*.md"]`

Neither approach causes hidden directories to be scanned.

## Environment

- rumdl version: 0.0.156
- OS: macOS (Darwin 25.0.0)

## Suggested Solution

Consider one of:
1. Include hidden directories by default (like most linters such as `ruff`, `eslint`, etc.)
2. Add a flag like `--all` or `--include-hidden` to opt into scanning hidden directories
3. At minimum, document this behavior clearly so users know they must explicitly specify hidden directories

## Workaround

For now, explicitly listing the hidden directory works:
```bash
rumdl check . .documentation
```

**Comment by davidpoblador** [comment]
Found the root cause! In `src/main.rs:602`, there's this line:

```rust
walk_builder.hidden(true); // Keep hidden files ignored unconditionally
```

The `ignore` crate's API works like this:
- `hidden(true)` = skip hidden files/directories
- `hidden(false)` = include hidden files/directories

The simple fix is to change that single line to:
```rust
walk_builder.hidden(false);
```

This will make `rumdl check` scan hidden directories by default, no new flags needed. 

I'm curious if there was a particular reason for excluding hidden directories in the first place? Just want to make sure we're not missing something before making the change.

Testing this change now.

**Comment by davidpoblador** [comment]
Created PR #103 with the fix. Tested locally and working as expected.

**Comment by rvben** [comment]
Hi @davidpoblador,

Thanks for raising this issue. In my projects hidden directories like .venv, .git, are typically not included in linting. Maybe that's why it made sense to just disable all hidden directories by default. But I recognize that this was a wrong assumption. It's indeed better to explicitly put them in .gitignore or exclude them from linting in the config, instead of making such assumptions from the tool.

The PR looks good, I'll merge it in. Thanks again! 🚀

**Comment by davidpoblador** [comment]
Thanks for being so swift. Also, thanks for `rumdl`!
